config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadata.java

```diff
@@ -0,0 +1,157 @@
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.bundle.builder;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+@JsonPropertyOrder({"id", "name", "groupName", "version", "type", "tags", "description", "reusable", "redeployable",
+        "hasRouting", "environmentIncluded", "definedEntities", "environmentEntities", "dependencies"})
+public class BundleMetadata implements Metadata {
+    private String type;
+    private String name;
+    private String id;
+    private String version;
+    private String groupName;
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    private String description;
+    private Collection<Metadata> definedEntities;
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    private Collection<String> tags;
+    private boolean reusable;
+    private boolean redeployable;
+    private boolean hasRouting;
+    private boolean environmentIncluded;
+    private Collection<Metadata> environmentEntities;
+
+    private BundleMetadata(String type, String id, String name, String groupName, String version) {
+        this.id = id;
+        this.type = type;
+        this.name = name;
+        this.groupName = groupName;
+        this.version = version;
+    }
+
+    @Override
+    public String getType() {
+        return type;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getVersion() {
+        return version;
+    }
+
+    public String getGroupName() {
+        return groupName;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public Collection<Metadata> getDefinedEntities() {
+        return definedEntities;
+    }
+
+    public Collection<String> getTags() {
+        return tags;
+    }
+
+    public boolean isReusable() {
+        return reusable;
+    }
+
+    public boolean isRedeployable() {
+        return redeployable;
+    }
+
+    public boolean isHasRouting() {
+        return hasRouting;
+    }
+
+    public boolean isEnvironmentIncluded() {
+        return environmentIncluded;
+    }
+
+    public Collection<Metadata> getEnvironmentEntities() {
+        return environmentEntities;
+    }
+
+    public static class Builder {
+        private final String id;
+        private final String name;
+        private final String type;
+        private String groupName;
+        private final String version;
+        private String description;
+        private boolean reusable;
+        private boolean redeployable;
+        private Collection<String> tags;
+        private Collection<Metadata> definedEntities = new LinkedList<>();
+        private Collection<Metadata> environmentEntities = new LinkedList<>();
+        private Collection<Metadata> dependencies = new LinkedList<>();
```

Comment:
is this value used ?

Reply:
Not now. We have not implemented bundle having dependency over another bundle.